rama-net/src/client/pool.rs

```diff
@@ -131,13 +132,59 @@ struct PooledConnection<C, ID> {
     last_used: Instant,
 }
 
+pub type FiFoReuseLruDropPool<C, ID> = LruDropPool<FiFoReuse, C, ID>;
+pub type RoundRobinReuseLruDropPool<C, ID> = LruDropPool<RoundRobinReuse, C, ID>;
+
 /// Connection pool that uses FiFo for reuse and LRU to evict connections
-pub struct FiFoReuseLruDropPool<C, ID> {
+pub struct LruDropPool<R, C, ID> {
     storage: Arc<Mutex<VecDeque<PooledConnection<C, ID>>>>,
     total_slots: Arc<Semaphore>,
     active_slots: Arc<Semaphore>,
     idle_timeout: Option<Duration>,
     returner: ConnReturner<C, ID>,
+    _marker: PhantomData<R>,
+}
+
+pub struct FiFoReuse;
```

Comment:
Implement `Clone`, `Debug` and `Default` for these please.
Also mark them non Exhaustive. You can Add also a `pub fn new() -> Self` for it.

This way we can still add state to it in future if we ever need to do so.